packages/office-ui-fabric-react/src/components/CommandBar/CommandBar.Props.ts

```diff
@@ -16,39 +20,74 @@ export interface ICommandBarProps extends React.HTMLAttributes<HTMLDivElement> {
   componentRef?: (component: ICommandBar) => void;
 
   /**
-   * Whether or not the search box is visible
-   * @defaultvalue false
+   * Items to render
    */
-  isSearchBoxVisible?: boolean;
+  items: ICommandBarItemProps[];
+
+  /**
+ * Items to render on the right side (or left, in RTL).
+ */
+  farItems?: ICommandBarItemProps[];
 
   /**
-   * Placeholder text to display in the search box
+   * Default items to have in the overflow menu
    */
-  searchPlaceholderText?: string;
+  overflowItems?: ICommandBarItemProps[];
 
   /**
-   * Items to render
+   * Text to be read by screen readers if there are overflow items and focus is on elipsis button
    */
-  items: IContextualMenuItem[];
+  elipisisAriaLabel?: string;
 
   /**
-   * Default items to have in the overflow menu
+  * Text to be read by screen readers if there are overflow items and focus is on elipsis button
+  */
+  elipisisIconProps?: IIconProps;
+
+  /**
+   * Custom styles to be mixed into individual button styles
    */
-  overflowItems?: IContextualMenuItem[];
+  buttonStyles?: IButtonStyles;
 
   /**
-   * Text to be read by screen readers if there are overflow items and focus is on elipsis button
+   * Custom render function for all non contextual menu buttons.
    */
-  elipisisAriaLabel?: string;
+  onRenderButton?: (item: ICommandBarItemProps) => JSX.Element;
+
+  /**
+   * Custom function to reduce data if items do not fit in given space. Return `undefined` if no more steps can be taken to avoid infinate loop.
+   */
+  onReduceData?: (data: ICommandBarData) => ICommandBarData;
 
   /**
-   * Items to render on the right side (or left, in RTL).
+   * Custom function to grow data if items are too small for the given space. Return `undefined` if no more steps can be taken to avoid infinate loop.
    */
-  farItems?: IContextualMenuItem[];
+  onGrowData?: (data: ICommandBarData) => ICommandBarData;
 
   /**
    * Additional css class to apply to the command bar
    * @defaultvalue undefined
    */
   className?: string;
-}
\ No newline at end of file
+}
+
+export interface ICommandBarItemProps extends IContextualMenuItem {

```

Comment:
We should also only use cacheKeys when the consumer has explicitly provided them. Unless you think we can compute them accurately in fabric.